lib/new_relic/agent/llm.rb

```diff
@@ -3,9 +3,47 @@
 # frozen_string_literal: true
 
 require_relative 'llm/llm_event'
-require_relative 'llm/chat_completion'
 require_relative 'llm/chat_completion_message'
 require_relative 'llm/chat_completion_summary'
 require_relative 'llm/embedding'
-require_relative 'llm/feedback'
 require_relative 'llm/response_headers'
+
+module NewRelic
+  module Agent
+    class LLM
+      INPUT = 'input'
+      CONTENT = 'content'
+      SEGMENT_PATTERN = %r{Llm/.*/OpenAI/.*}.freeze
```

Comment:
I think we can be stricter with the pattern, especially doing so results in a performance boost. One idea is to not use `*`:

```suggestion
      SEGMENT_PATTERN = %r{Llm/.+/OpenAI/.+}.freeze
```

Reply:
So the `*` is saying anything cane come before/after `/OpenAI/`, but what is the `+` saying?

Reply:
> So the * is saying anything

`*` = 0 or more
`+` = 1 or more

The existing pattern matches `'Llm//OpenAI/'` with nothing between the `Llm` and `OpenAI` bits and nothing following the 'OpenAI' bit. Switching to `.+` to ensure something is present in both places could make it a little better. I'd need to have a better understanding of the type of data that is expected to be present but not match in order to suggest a better pattern.

Reply:
There are only two segment names as of right now that we are expecting: `Llm/embedding/OpenAI/embeddings` and `Llm/completion/OpenAI/chat`, so your suggested pattern works ◡̈